dlib/dnn/loss_abstract.h

```diff
@@ -769,6 +769,68 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+    // In semantic segmentation, 65535 classes ought to be enough for anybody.
+    typedef unsigned short matrixoutput_label_t;
```

Comment:
I would just use uint16 for this and not even bother with a typedef to give it a special name.